client/js/color.js

```diff
@@ -0,0 +1,57 @@
+export function toHex(inputColor) {
+  const hexPattern = /^#([A-Fa-f0-9]{6})$/;
+  if (hexPattern.test(inputColor)) {    
+    return inputColor;
+  }
+  const canvas = document.createElement('canvas');
+  const ctx = canvas.getContext('2d');
+  ctx.fillStyle = inputColor;
+  const hexColor = ctx.fillStyle;  
+  canvas.remove();
+  return hexColor;
+}
+
+export function toRGB(inputColor) {
+  let hex = toHex(inputColor)
+  hex = hex.slice(1);
+  var rgb = hex.match(new RegExp('(.{' + hex.length/3 + '})', 'g')).map(function(l) {
+    return parseInt(hex.length%2 ? l+l : l, 16);
+  });
+  return `rgb(${rgb.join(',')})`;    
+}
+
+export function calcLuminance(inputColor) {
+  let color = toHex(inputColor);
+  const r = parseInt(color.slice(1, 3), 16);
+  const g = parseInt(color.slice(3, 5), 16);
+  const b = parseInt(color.slice(5, 7), 16);
+  const luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255;
```

Comment:
Should we apply sRGB gamma correction as [defined in WCAG 2.x](https://www.w3.org/WAI/GL/wiki/Relative_luminance)?

Reply:
I'm not sure if I understand the difference.  The notes so there is no actual difference in the 8-bit colors we are using.  So I changed the formula to be what I think WCAG 3 is saying.  I got the same results (as expected).  Is this what you menat?

Reply:
The gamma correction is this part:
```
if RsRGB <= 0.03928 then R = RsRGB/12.92 else R = ((RsRGB+0.055)/1.055) ^ 2.4
if GsRGB <= 0.03928 then G = GsRGB/12.92 else G = ((GsRGB+0.055)/1.055) ^ 2.4
if BsRGB <= 0.03928 then B = BsRGB/12.92 else B = ((BsRGB+0.055)/1.055) ^ 2.4
```

Reply:
Okay.  I think I got it.  But I need to change the threshold.